pandas/io/formats/format.py

```diff
@@ -1507,11 +1507,28 @@ def get_result_as_array(self) -> np.ndarray:
         the parameters given at initialisation, as a numpy array
         """
 
-        def format_with_na_rep(values: ArrayLike, formatter: Callable, na_rep: str):
+        def format_with_na_rep(
+            values: ArrayLike,
+            formatter: Callable,
+            na_rep: str,
+            is_complex: bool = False,
+        ):
+            def _format_complex_na(val, fmt) -> str:
```

Comment:
Instead of a nested function here did you give any consideration to updating the `_value_formatter` method? Wondering if it is cleaner to put this code there

Reply:
Wasn't sure if passing is_complex in there is better, but no strong preference on my side

Reply:
Yeah I would slightly prefer for `formatter` knowing how to format this instead of bypassing it